src/coreclr/inc/corinfo.h

```diff
@@ -3167,6 +3193,32 @@ class ICorDynamicInfo : public ICorStaticInfo
                     void                  **ppIndirection = NULL
                     ) = 0;
 
+    //------------------------------------------------------------------------------
+    // getReadonlyStaticFieldValue: returns true and the actual field's value if the given
+    //    field represents a statically initialized readonly field of any type, it might be:
+    //    * integer/floating point primitive
+    //    * null
+    //    * frozen object which in turn can be:
+    //       * string
+    //       * RuntimeType (CoreCLR only)
+    //       * Array (NativeAOT only)
+    //       * Delegate (NativeAOT only)
+    //       * Object wihtout fields (NativeAOT only)
```

Comment:
```suggestion
    //    * frozen object reference (string, array or object)
```

It can be object with fields too.

I do not think it is necessary to enumerate all cases in the comment. It included everything (modulo runtime details) and the runtime details are going to change over time.

Reply:
Good point, I already plan to experiment with some structs here e.g. DateTime/Guid 🙂 (unrelated to FOH)